unity-client/Assets/Scripts/MainScripts/DCL/Controllers/StickersController/StickersController.cs

```diff
@@ -0,0 +1,20 @@
+﻿using UnityEngine;
+
+public class StickersController : MonoBehaviour
+{
+    private StickersFactory stickersFactory;
+
+    private void Awake() { stickersFactory = Resources.Load<StickersFactory>("StickersFactory"); }
```

Comment:
I have this conundrum:

- If this is a `MonoBehaviour`, why use `Resources.Load`? We can just reference the factory.
- However, it does make sense to make this a `MonoBehaviour`? The `transform` reference can be injected and doesn't need many others or listening to component events.

Reply:
We could indeed reference the factory, but since this script goes in multiple prefabs, I didnt want to bother with nuisances like missing references on any of them. I dont mind using serialization though.

I like this controller to be a `MonoBehaviour` because our Avatar is a bit of a mess currently. There's no "main" controller that instances the `AvatarRenderer` or the `AvatarAnimator`. Them both are autonomous and working as components so to be able to make the `StickersController` a POCO I would need somebody to instantiate. The `AvatarRenderer` might be a good candidate but I couldn't fit them properly since you can trigger stickers outside of the emotes flow.

Reply:
Thanks for the explanation. I agree we need some sort of extra layer that handles the complexity of the avatar. I'd like to discuss later if it would be more convenient to have a controller for all the avatar systems and for the expressions in general as well. I'm worried that the highest layer of abstraction for emotes is the `AvatarAnimator`, as it doesn't sound enough for now.

Going to approve for now but I think we have to improve the design for this stuff.

Reply:
Yeah the avatar refactoring has been on hold for too long IMHO